lib/isolator/isolate.rb

```diff
@@ -3,10 +3,46 @@
 module Isolator
   # Add .isolate function to build and register adapters
   module Isolate
+    TODO_PATH = ".isolator_todo.yml"
+
     def isolate(id, **options)
       raise "Adapter already registered: #{id}" if Isolator.adapters.key?(id.to_s)
       adapter = AdapterBuilder.call(options)
+      register_todos(id.to_s, adapter) if isolator_todos
       Isolator.adapters[id.to_s] = adapter
     end
+
+    private
+
+    def register_todos(id, adapter)
+      ignored_paths = isolator_todos.fetch(id, [])
+      ignored_paths.each { |path| register_todo(adapter, path) }
+    end
+
+    def register_todo(adapter, path)
+      ignored_files = Dir[Rails.root.join(path)]
+
+      if ignored_files.blank?
+        ignore_line(adapter, path)
+      else
+        ignore_in_files(adapter, ignored_files)
+      end
+    end
+
+    def ignore_line(adapter, line)
+      full_line = Rails.root.join(line).to_s
+      adapter.ignore_if { caller.any? { |caller_path| caller_path.include?(full_line) } }
+    end
+
+    def ignore_in_files(adapter, ignored_files)
+      adapter.ignore_if do
+        caller_paths = caller.map { |row| row.split(":in").first }
```

Comment:
What about using Regex here (build from ignore pattern) to avoid splitting?

Then we could build the same `ignore_if` block for both exact lines and masks.